### PR TITLE
adding Sixel Support to PwshSpectreConsole

### PR DESCRIPTION
**Pull request by trackd**
Sixel

image

still needs a bit of work but most things seem to render ok..

need to investigate transparency a bit more.

### CODE REVIEW

File: PwshSpectreConsole/private/classes/PwshSpectreConsole.Sixel.cs [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+          // this is the last pixel in the row.. i think its needed. it looks weird i know.
+          if (repeatCounter > 1)
+          {
+            writer.Write($"#{lastPixel}!{repeatCounter}{c}");
+          }
+          else
+          {
+            writer.Write($"#{lastPixel}{c}");
+          }
+          writer.Write("$");
```
Comment (ShaunLawrie) [inline PwshSpectreConsole/private/classes/PwshSpectreConsole.Sixel.cs:80]: It'd be worth dragging out the magic characters into some const values with names like SIXEL_CR / SIXEL_LF or something so it makes more sense what these are.

**Comment by trackd** [comment]
```md
| Method        | Mean     | Error    | StdDev   | Median   | Gen0      | Gen1      | Gen2      | Allocated |
|-------------- |---------:|---------:|---------:|---------:|----------:|----------:|----------:|----------:|
| ImgToSixelOld | 34.86 ms | 0.693 ms | 1.250 ms | 34.25 ms | 1500.0000 | 1031.2500 | 1000.0000 |   7.57 MB |
| ImgToSixelNew | 27.59 ms | 0.516 ms | 0.483 ms | 27.44 ms |  968.7500 |  968.7500 |  968.7500 |   1.19 MB |

```

converted it to stringbuilder.. seems alot better :)
also refactored the class a bit, it's not broken up into much smaller methods which will simplify when adding gif/animated sixels.

**Comment by ShaunLawrie** [comment]
Most of this code is upstream in https://github.com/spectreconsole/spectre.console/compare/main...ShaunLawrie:spectre.console:main

It's in prerelease atm (just copied the dlls into the repo) and it's full of hacks and visual bugs lol https://prerelease.pwshspectreconsole.com/reference/images/get-spectresixelimage/

image

**Comment by trackd** [comment]
closing this as we broke it out into it's own [sixel module](https://github.com/trackd/sixel) and adding the code upstream to SpectreConsole.

will push the non-sixel stuff in a separate pr

**Pull request closed without merging**